build: exclude alpha APIs from review requirement

jason-ha opened the pull request:
Whitelist `public`, `beta`, and `legacy.*` as requiring review.

`build-tools` and `server` exclusions can be removed as not broken into those categories.

jason-ha committed:
build: remove experimental presence package from API review

anthony-murphy commented on .github/CODEOWNERS:
```diff
@@ -59,6 +59,7 @@
 # TODO: if we ever add `.internal` API reports, we will want to omit them here
 /**/api-report/*.api.md                        @microsoft/fluid-cr-api
 /build-tools/**/api-report/*.api.md            # Do not require API review for build-tools packages
+/packages/framework/presence/**/api-report/*.api.md # Do not require API review for experimental presence package
```
rather than this, can you just turn off api docs generation in the presence package?

jason-ha replied:
I could, but I prefer seeing what is really changing if anything in the report.

ChumpChief commented on .github/CODEOWNERS:
```diff
@@ -59,6 +59,7 @@
 # TODO: if we ever add `.internal` API reports, we will want to omit them here
 /**/api-report/*.api.md                        @microsoft/fluid-cr-api
```
If the idea of "alpha" is to move fast with ideas and try things out, maybe instead we should only require API review for *.public.api.md?  And maybe *.beta.api.md, *.legacy.alpha.api.md?

jason-ha replied:
I like this. New proposal.

anthony-murphy replied:
the thing to keep in mind is that this could result in a heavy lift where substantial changes are requested from alpha to beta, as we are essentially saying review is deferred until the beta phase.

anthony-murphy replied:
it would also be useful if the use case could be expanded upon in this review? For instance, what is the plan, and what about the plan makes the current state hard to deal with? Is there a possibility to work on this as internal, and only switch to alpha before release?

jason-ha replied:
I would say that review enforcement is deferred until beta. Hopefully APIs are getting reviewed once there is something worthy. Maybe that is alpha maybe that is later.
I certainly welcome and encourage looking at the presence API, but that doesn't need to happen at PR time.
I defer to API council for policy they want.

jason-ha committed:
build: require review for select API levels
notably exclude alpha from requirement.
Whitelist of public, beta, and legacy cleans up TODO and some exclusions.

ChumpChief reviewed (approved):
I think this makes sense - @anthony-murphy any hesitations or does this seem ok to you too?
